Mathlib/Algebra/Group/Translate.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Algebra.BigOperators.Pi
+import Mathlib.Algebra.Group.Pi.Basic
+import Mathlib.Data.Set.Pointwise.SMul
+import Mathlib.GroupTheory.GroupAction.DomAct.Basic
+
+/-!
+# Translation operator
+
+This file defines the translation of a function from a group by an element of that group.
+
+## Notation
+
+`τ a f` is notation for `translate a f`.
+
+## See also
+
+Generally, translation is the same as acting on the domain by subtraction. This setting is
+abstracted by `DomAddAct` in such a way that `τ a f = DomAddAct.mk (-a) +ᵥ f` (see
+`translate_eq_domAddActMk_vadd`). Using `DomAddAct` is irritating in applications because of this
+negation appearing inside `DomAddAct.mk`. Although mathematically equivalent, the pen and paper
+convention is that translating is an action by subtraction, not by addition.
+-/
+
+open Function Set
+open scoped Pointwise
+
+variable {ι α β M G H : Type*} [AddCommGroup G]
```

Comment:
I think this can be useful in noncommutative contexts as well. So please generalize the definition.

Reply:
But then do you want left or right translation?

Reply:
right (pun intended), that's the question :grinning: 
We might want to have both. I always forget what left/right means: is the constant or the variable on the left/right?